### PR TITLE
Onboarding OAuth callback glitch

### PR DESCRIPTION
**Pull request by KATT**
## Issue showcased

https://user-images.githubusercontent.com/459267/139540804-f8415a07-43ba-44e7-9acb-220b4e18a99c.mov

## Fixes

- Fixes Google + Microsoft integration callback glitches
- Needs testing

## Testing

- [x] Google Calendar connect / disconnect
- [x] Microsoft connect / disconnect

### CODE REVIEW

File: components/integrations/ConnectIntegrations.tsx [hunk 1]
```diff
@@ -13,13 +14,20 @@ export default function ConnectIntegration(props: {
...
+    const stateStr = encodeURIComponent(JSON.stringify(state));
+    const searchParams = `?state=${stateStr}`;
+    const res = await fetch("/api/integrations/" + type.replace("_", "") + "/add" + searchParams);
     if (!res.ok) {
       throw new Error("Something went wrong");
     }
     const json = await res.json();
-    window.location.href = json.url;
+    const { url } = json;
+    window.location.href = url;
```
Comment (KATT) [inline components/integrations/ConnectIntegrations.tsx:30]:
```suggestion
    window.location.href = json.url;
```

**Review by zomars — APPROVED** [review]
Yup this should do

**Comment by emrysal** [comment]
It works.. but I am wondering if this can't simply be solved by doing the integration connection in a `_blank` - that way we don't have to faff about 🤔

**Comment by PeerRich** [comment]
this ready to be merged @KATT ?

**Comment by KATT** [comment]
> this ready to be merged @KATT ?

cc @PeerRich
Needs testing on Microsoft yet - I don't have credentials for that, will ping @baileypumfleet on Slack

**Pull request merged**